platforms/jvm/testing-jvm/src/integTest/groovy/org/gradle/testing/junit/junit4/AbstractJUnit4LoggingOutputCaptureIntegrationTest.groovy

```diff
@@ -83,79 +82,88 @@ abstract class AbstractJUnit4LoggingOutputCaptureIntegrationTest extends Abstrac
+            "Test anotherOk(OkTest) -> after out\n" +
+            "Test anotherOk(OkTest) -> after err\n" +
+            "Test ok(OkTest) -> test constructed\n" +
+            "Test ok(OkTest) -> before out\n" +
+            "Test ok(OkTest) -> before err\n" +
+            "Test ok(OkTest) -> test out: \u03b1</html>\n" +
+            "Test ok(OkTest) -> test err\n" +
+            "Test ok(OkTest) -> after out\n" +
+            "Test ok(OkTest) -> after err\n" +
+            "Test class OkTest -> after class out\n" +
+            "Test class OkTest -> after class err\n"
+        )
+
+        // This test covers current behaviour, not necessarily desired behaviour
+
+        def xmlReport = new JUnitXmlTestExecutionResult(testDirectory)
+        def classResult = xmlReport.testClass("OkTest")
+        classResult.assertTestCaseStdout("ok", is(
+            "test constructed\n" +
+            "before out\n" +
+            "test out: \u03b1</html>\n" +
+            "after out\n"
+        ))
+        classResult.assertTestCaseStderr("ok", is(
+            "before err\n" +
+            "test err\n" +
+            "after err\n"
+        ))
+        classResult.assertStdout(is(
+            "class loaded\n" +
+            "before class out\n" +
+            "after class out\n"
+        ))
+        classResult.assertStderr(is(
+            "before class err\n" +
+            "after class err\n"
+        ))
+
+        def results = resultsFor(testDirectory)
+        def classReport = results.testPath("OkTest").onlyRoot()
         classReport.assertStdout(is(
             "class loaded\n" +
             "before class out\n" +
+            "after class out\n"
+        ))
+        classReport.assertStderr(is(
+            "before class err\n" +
+            "after class err\n"
+        ))
+        def okReport = results.testPath("OkTest", "ok").onlyRoot()
+        okReport.assertStdout(is(
             "test constructed\n" +
             "before out\n" +
-            "ok out\n" +
-            "after out\n" +
+            "test out: \u03b1</html>\n" +
+            "after out\n"
+        ))
+        okReport.assertStderr(is(
+            "before err\n" +
+            "test err\n" +
+            "after err\n"
+        ))
+        def anotherOkReport = results.testPath("OkTest", "anotherOk").onlyRoot()
+        anotherOkReport.assertStdout(is(
             "test constructed\n" +
             "before out\n" +
-            "test out: \u03b1</html>\n" +
-            "after out\n" +
-            "after class out\n"
+            "ok out\n" +
+            "after out\n"
         ))
-        classReport.assertStderr(is(
-            "before class err\n" +
+        anotherOkReport.assertStderr(is(
             "before err\n" +
             "ok err\n" +
-            "after err\n" +
-            "before err\n" +
-            "test err\n" +
-            "after err\n" +
-            "after class err\n"
+            "after err\n"
```

Comment:
The output here differs because it's actually reported more granularly now.  We should note this in the release notes.